src/java.base/share/classes/jdk/internal/foreign/StringSupport.java

```diff
@@ -93,10 +118,45 @@ private static void writeFast_int(MemorySegment segment, long offset, Charset ch
         segment.set(JAVA_INT, offset + bytes.length, 0);
     }
 
-    private static int strlen_byte(MemorySegment segment, long start) {
+    private static int native_strlen_byte(MemorySegment segment, long start) {
+        try {
```

Comment:
I think this `try` block is only meant to deal with a potential exception from `invokeExact`? But, since it covers many lines of code, it's hard to tell what it's doing.

I suggest trying to limit the covered lines, for example by creating out-of-line wrapper methods for `STRNLEN` and `STRNLEN_TRIVIAL` and then putting the try/catch only in there.